From f83e6626e9abb8c9e23cca881af85b288db6e45d Mon Sep 17 00:00:00 2001
From: Ayose <ayosec@gmail.com>
Date: Thu, 19 Dec 2019 15:52:09 +0000
Subject: [PATCH] Show full contents when an unmatched character is found

---
 include/cairo/context.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/cairo/context.hpp b/include/cairo/context.hpp
index 8ed4f50..29e89a7 100644
--- a/include/cairo/context.hpp
+++ b/include/cairo/context.hpp
@@ -218,7 +218,7 @@ namespace cairo {
 
         char unicode[6]{'\0'};
         utils::ucs4_to_utf8(unicode, chars.begin()->codepoint);
-        m_log.warn("Dropping unmatched character %s (U+%04x)", unicode, chars.begin()->codepoint);
+        m_log.warn("Dropping unmatched character %s (U+%04x) in '%s'", unicode, chars.begin()->codepoint, string(t.contents));
         utf8.erase(chars.begin()->offset, chars.begin()->length);
         for (auto&& c : chars) {
           c.offset -= chars.begin()->length;
-- 
2.20.1

